Components/Chat/ChatRoom.tsx

```diff
@@ -0,0 +1,196 @@
+			setChatUsers(chatUsers.users);
+		};
+
+		fetchChatUsers();
+	}, [accessToken, chatId]);
+
+	// 'fetch-messages' 이벤트 등록
+	useEffect(() => {
+		if (socket?.connected) {
+			socket.emit('fetch-messages');
+
+			socket.on('messages-to-client', (messagesObject) => {
+				setMessages(messagesObject.messages);
+			});
+		}
+
+		return () => {
+			socket?.off('messages-to-client');
+		};
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [socket.connected]);
+
+	// message-to-client 이벤트 등록
+	useEffect(() => {
+		if (socket?.connected) {
+			socket.on('message-to-client', (responseData) => {
+				setMessages((prevMessages) => [...prevMessages, responseData]);
+			});
+		}
+
+		return () => {
+			socket?.off('message-to-client');
+		};
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [socket.connected]);
+
+	// join 이벤트 등록
+	useEffect(() => {
+		if (socket?.connected) {
+			socket.on('join', (responseData) => {
+				setChatUsers([...chatUsers, ...responseData.users]);
+
+				if (responseData.users[0] !== userId) {
+					Swal.fire(`${responseData.leaver} 님이 퇴장하셨습니다.`);
+				}
+			});
+		}
+
+		return () => {
+			socket.off('join');
+		};
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [chatUsers, socket.connected]);
+
+	// leave 이벤트 등록
+	useEffect(() => {
+		if (socket?.connected) {
+			socket.on('leave', (responseData) => {
+				setChatUsers([...chatUsers, ...responseData.users]);
+
+				if (responseData.leaver !== userId) {
+					Swal.fire(`${responseData.leaver} 님이 퇴장하셨습니다.`);
+				}
+			});
+		}
+
+		return () => {
+			socket.off('leave');
+		};
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [chatUsers, socket.connected]);
+
+	// 서버로 메시지 전송
+	const sendMessage = () => {
+		if (newMessage.trim() !== '') {
+			socket?.emit('message-to-server', newMessage);
+			setNewMessage('');
+		}
+	};
+
+	// enter 입력 시 메시지 전송
+	const handleKeyPress = (e: React.KeyboardEvent<HTMLInputElement>) => {
+		if (e.key === 'Enter') {
+			sendMessage();
+		}
+	};
+
+	return (
+		<div className="h-screen overflow-y-scroll">
+			<ChatHeader
+				chatId={chatId}
+				chatName={chatName}
+				chatUsers={chatUsers.length}
+			/>
+			{privateValue === 'true' ? (
+				<>
+					{chatUsers.length === 2 || chatUsers.length === 1 ? (
+						<>
+							<RenderChats
+								messages={messages}
+								chatUsers={chatUsers}
+								useModal={false}
+							/>
+						</>
+					) : (
+						<>
+							<RenderChats
+								messages={messages}
+								chatUsers={chatUsers}
+								useModal={true}
+							/>
+						</>
+					)}
+				</>
+			) : (
+				<>
+					<RenderChats
+						messages={messages}
+						chatUsers={chatUsers}
+						useModal={false}
+					/>
+				</>
+			)}
+
+			<div className="w-full sm:w-[425px] h-14 flex justify-evenly items-center py-8 bg-footer fixed bottom-0">
+				<input
+					type="text"
+					value={newMessage}
+					onChange={(e) => setNewMessage(e.target.value)}
+					onKeyPress={handleKeyPress}
```

Comment:
이벤트를 전달하고 setNewMessage 함수에서 target.value로 접근했다면 onChange에 콜백을 전달할 때 불필요한 익명함수 선언을 막을 수 있었을 것 같습니다.

```suggestion
					onChange={setNewMessage}
					onKeyPress={handleKeyPress}
```